numpy/core/src/common/dlpack/dlpack.h

```diff
@@ -0,0 +1,199 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file dlpack.h
+ * \brief The common header of DLPack.
+ */
```

Comment:
This should include a commit hash from upstream and probably a URL.

What's the rationale behind everyone having their own copy of this header rather than taking `dlpack` as a dependency?

Reply:
cc @rgommers I think you can better answer this one.

Reply:
> This should include a commit hash from upstream and probably a URL.

That sounds like a good idea.

> What's the rationale behind everyone having their own copy of this header rather than taking dlpack as a dependency?

I think adding a dependency is simply a much larger ask than vendoring a ~200 LoC header. Right now the only runtime dependencies we have are Python itself, and a BLAS library. And build-time dependencies Python/BLAS/Cython/compilers. Adding anything both makes things more fragile, involves extra work for packagers, and introduces coordination issues when the dependency needs updating.